lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2285,34 +2306,19 @@ GetFollowAsyncContextUnwindPlan(RegisterContext *regctx, ArchSpec &arch,
   const int32_t ptr_size = 8;
   row->SetOffset(0);
 
-  uint32_t async_context_regnum;
-  uint32_t fp_regnum;
-  uint32_t pc_regnum;
-  uint32_t dummy_regnum;
-
-  if (arch.GetMachine() == llvm::Triple::x86_64) {
-    async_context_regnum = dwarf_r14_x86_64;
-    fp_regnum = dwarf_rbp_x86_64;
-    pc_regnum = dwarf_rip_x86_64;
-    dummy_regnum = dwarf_r15_x86_64;
-  } else if (arch.GetMachine() == llvm::Triple::aarch64) {
-    async_context_regnum = arm64_dwarf::x22;
-    fp_regnum = arm64_dwarf::fp;
-    pc_regnum = arm64_dwarf::pc;
-    dummy_regnum = arm64_dwarf::x23;
-  } else {
+  auto registers = GetAsyncUnwindRegisters(arch.GetMachine());
+  if (!registers)
     return UnwindPlanSP();
-  }
 
   // In the general case, the async register setup by the frame above us
   // should be dereferenced twice to get our context, except when the frame
   // above us is an async frame on the OS stack that takes its context directly
   // (see discussion in GetRuntimeUnwindPlan()). The availability of
   // dummy_regnum is used as a marked for this situation.
-  if (regctx->ReadRegisterAsUnsigned(dummy_regnum, (uint64_t)-1ll) !=
+  if (regctx->ReadRegisterAsUnsigned(registers->dummy_regnum, (uint64_t)-1ll) !=
       (uint64_t)-1ll) {
-    row->GetCFAValue().SetIsRegisterDereferenced(async_context_regnum);
-    row->SetRegisterLocationToSame(async_context_regnum, false);
+    row->GetCFAValue().SetIsRegisterDereferenced(registers->async_ctx_regnum);
+    row->SetRegisterLocationToSame(registers->async_ctx_regnum, false);
```

Comment:
most of the change in this PR is noise, where regular local variables are now fields on structs. I could unpack the struct and return these all to using local vars, if preferred.